.gitignore

```diff
@@ -19,3 +18,6 @@ obj
 
 .idea
 .DS_Store
+
+template
```

Comment:
Are we sure that `template` folder should be in `.gitignore`? We make changes to the files in this folder when we need to update our officially supported templates

Reply:
Do you know how that got there? We have code that updates the .gitignore if you run `faas-cli new`

Reply:
I guess running `faas-cli new --lang dockerfile doc1` in your project repo rather than a distant `function` folder would update the project git repo, rather than the local function repo.

Reply:
I second what @rgee0 said. In the `faas-cli new` command we will always create `.gitignore` which has `template` and `build` added to the ignore list, or if `.gitignore` exists, it will add `template` and `build` to it.

Like Richard said, because the `template` dir was added to `.gitignore`, the `Dockerfile/template.yml` file was ignored.